Join PipeHistory queries with pipes for additional tags

yogevyuval opened the pull request:
### What does this PR do?
This PR adds 2 additional tags to the PipeHistory query

### Motivation
Without these tags, these metrics are almost useless. Imagine the following:
2 schemas, one is dev env, one is production. Each schema has a pipe named “INPUT_PIPE” that ingests data into snowflake.
Without these tags, the metric is useless, because it aggregates across non related schemas.


- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] [Changelog entries](https://datadoghq.dev/integrations-core/guidelines/pr/#changelog-entries) must be created for modifications to shipped code
- [ ] Add the `qa/skip-qa` label if the PR doesn't need to be tested during QA.

yogevyuval commented:
@hestonhoffman Thanks
First contriubtion for me here - how do I fix the changelog parts? + not sure about the tests here, will they fail if the query is wrong?

hestonhoffman commented:
> @hestonhoffman Thanks First contriubtion for me here - how do I fix the changelog parts? + not sure about the tests here, will they fail if the query is wrong?

That's a good question for @DataDog/agent-integrations :)

iliakur commented on snowflake/CHANGELOG.md:
```diff
@@ -2,6 +2,12 @@
+
+***Added***:
+
+* Added `pipe_schema` and `pipe_catalog` tags to snowpipe metrics. ([#16222](https://github.com/DataDog/integrations-core/pull/16222))
```
@yogevyuval would you mind undoing this change, updating your `ddev` to the latest version and running `ddev release changelog new`? We recently reworked the way we generate changelogs.

yogevyuval replied:
@iliakur Done. Can you take a look again?
Also - anyone can help with validating and tests?

iliakur commented on ddev/changelog.d/16222.added:
Hmm, not sure why this file got added. Was this the result of running the ddev command?

Could you post the output of `git status` here? After we remove this file, the PR lgtm!

yogevyuval replied:
Looks like a mistake, i removed it.
Any way to confirm my query does not break anything?

yogevyuval commented:
@iliakur What's next needed here? Anyway to validate the query?

iliakur commented:
@yogevyuval sorry about the delay. I'm about to go on holiday, my colleague @alopezz will take over next week. Have a good weekend and thanks again for your patience!

alopezz commented on snowflake/datadog_checks/snowflake/queries.py:
```diff
@@ -191,12 +191,16 @@
-        'select pipe_name, avg(credits_used), sum(credits_used), avg(bytes_inserted), sum(bytes_inserted), '
-        'avg(files_inserted), sum(files_inserted) from pipe_usage_history '
-        'where start_time >= DATEADD(hour, -24, current_timestamp()) group by 1;'
+        'select history.pipe_name,p.pipe_schema, p.pipe_catalog, '
```
Style nit
```suggestion
        'select history.pipe_name, p.pipe_schema, p.pipe_catalog, '
```

alopezz commented on snowflake/datadog_checks/snowflake/queries.py:
```diff
@@ -191,12 +191,16 @@
     ),
     'columns': [
         {'name': 'pipe', 'type': 'tag'},
+        {'name': 'pipe_schema', 'type': 'tag'},
+        {'name': 'pipe_catalog', 'type': 'tag'},
```
Assuming that this do refer to these concepts, maybe it would make sense to use the same tags that are used for other metrics? That should make it easier to cross-reference.

```suggestion
        {'name': 'schema', 'type': 'tag'},
        {'name': 'database', 'type': 'tag'},
```

alopezz reviewed (commented):
Sadly, we don't have an automated way of testing the integration against an actual instance. If you want, you can add a test with a mocked response such as those on https://github.com/DataDog/integrations-core/blob/master/snowflake/tests/test_queries.py, but since I don't see one already in place for this query, I'm not going to block the PR based on that.

I'm going to test the PR out manually to make sure the query is valid before approval.

yogevyuval commented:
> Sadly, we don't have an automated way of testing the integration against an actual instance of Snowflake. If you want, you can add a test with a mocked response such as those on https://github.com/DataDog/integrations-core/blob/master/snowflake/tests/test_queries.py, but since I don't see one already in place for this query, I'm not going to block the PR based on that.
> 
> I'm going to test the PR out manually to make sure the query is valid before approval.

Fixed the PR comments (and adjusted the changelog entry accordingly).
Appreciate you helping out with manual testing - let me know if everything worked our or if additional changes are needed!

yogevyuval commented:
@alopezz Did you have a chance to run this? Anything else from my side needed?

alopezz commented:
Sorry for the delay. I was trying to get the necessary data into a test account to get some data from the modified query but I haven't been able to do so. The query itself is fine in that it doesn't produce an error, I'm simply not getting any results for extra confirmation.

Would you be able to share some example results of that query? Since you've found the need to improve the query I suppose you should be able to get some data from it. I can then do some final testing out of that.

yogevyuval commented:
> Sorry for the delay. I was trying to get the necessary data into a test account to get some data from the modified query but I haven't been able to do so. The query itself is fine in that it doesn't produce an error, I'm simply not getting any results for extra confirmation.

@alopezz 
Yes, I ran this query on my account and it worked well. The reason I was a bit worried is actually things like specifying the database + schema before the table name or not, because for example in the original code it uses FROM pipe_history without any db or schema, and in the join I added I specified snowflake.account_usage.pipes.

So I wanted to make the query actually compiles well, no permission errors or other things like that

yogevyuval commented:
> > @alopezz Yes, I ran this query on my account and it worked well. The reason I was a bit worried is actually things like specifying the database + schema before the table name or not, because for example in the original code it uses FROM pipe_history without any db or schema, and in the join I added I specified snowflake.account_usage.pipes.
> > So I wanted to make the query actually compiles well, no permission errors or other things like that
> 
> I see what you mean (I hadn't payed close enough attention to that); technically it shouldn't matter because that query is always meant to be run against the `SNOWFLAKE` database and is only available when the schema is set to `ACCOUNT_USAGE`(see [config](https://github.com/DataDog/integrations-core/blob/706cb4b8edef06f1a9ee43647a81a03f9edef8af/snowflake/datadog_checks/snowflake/data/conf.yaml.example#L155-L166)), so both qualifying and not qualifying the table name works. You might want to remove the qualification from the reference to the `pipes` for consistency with the rest of the existing queries, other than that I consider this ready to merge.

@alopezz I removed the qualifications

alopezz reviewed (approved):
Nice, thanks for your patience!
